pkg/adapters/grpc/errors_test.go

```diff
@@ -55,4 +68,47 @@ var _ = Describe("GRPC Errors", func() {
 			})
 		})
 	})
+
+	Context("Logging Arg", func() {
+		When("Sensitive Fields", func() {
+			It("Should contain sensitive fields", func() {
+				v1 := grpc.LogArg("string")
+				Expect(v1).To(BeEquivalentTo("string"))
+
+				v2 := grpc.LogArg(123)
+				Expect(v2).To(BeEquivalentTo("123"))
+
+				v3 := grpc.LogArg(true)
+				Expect(v3).To(BeEquivalentTo("true"))
+
+				v4 := grpc.LogArg(12.3)
+				Expect(v4).To(BeEquivalentTo("12.3"))
+
+				type SecretValue struct {
+					Type   string `log:"Type"`
+					Factor int    `log:"-"`
+					Value  string
+				}
+
+				type Secret struct {
+					Name    string       `json:"Name" log:"Name"`
+					Version string       `json:"Version" log:"Version"`
+					Value   *SecretValue `json:"Value" log:"Value"`
+				}
+
+				data := Secret{
+					Name:    "name",
+					Version: "3",
+					Value: &SecretValue{
+						Type:   "key",
+						Factor: 2,
+						Value:  "2a4wijgPq0PpwJ76IjT7&lTBZ$5SGRcq",
+					},
+				}
+
+				v5 := grpc.LogArg(data)
+				Expect(v5).To(BeEquivalentTo("{Name: name, Version: 3, Value: {Type: key}}"))
+			})
+		})
+	})
 })
```

Comment:
Happy to help with this but the cases here should be more descriptive and granular.

> This is just an example, no need to use it verbatim
```go
Context("Logging Arg", func() {
  When("String", func() {
    testString := "test"
    It("Should return value", func() {
      Expect(grpc.LogArg(testString)).To(Equal("test"))
    })
  })

  When("Int", func() {
     ...
  })

  When("Struct", func() {
    ...
  })
  
  When("Nested Struct", func() {
    ...
  })
})
```